UsageStats: Move stats collection to separate service

sakjur opened the pull request:
**What this PR does / why we need it**:

Moves all the usage stats collection away from the usage stats service and into a service that only handles stats collection. Eventually, we'll want the services to own their own stats and register them as "external metrics", but that's hard to do until usage stats stops importing external services (risk of creating circular dependencies). This PR takes the first step splitting up the logic of sending and managing usage stats reports and collecting the data for the usage stats reports.

**Which issue(s) this PR fixes**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer**:

Most of the changes are "just" code that's moved from one place to another, it might be easier to review by looking at each commit individually.

sakjur committed:
Remove specific stats from usage stats service

sakjur committed:
Create statscollector service

sakjur committed:
refactor

sakjur committed:
Update and move tests

Mostly equivalent tests to before, but they've been divided over the two
services and removed the behavior driven legacy from GoConvey to
reduce the complexity of the tests.

bergquist reviewed (approved):
Looks good but there is also a fair amount of changes going on. Might be a good idea if someone else reviews this as well.

sakjur committed:
Merge branch 'main' into emil/stats-separate-collector

sakjur committed:
Merge branch 'main' into emil/stats-separate-collector

# Conflicts:
#	pkg/infra/usagestats/service/service.go
#	pkg/infra/usagestats/service/usage_stats.go
#	pkg/infra/usagestats/service/usage_stats_test.go

sakjur committed:
Collect featuremgmr metrics (copied over from #47407)

I removed the metrics registration from the feature manager in the merge
and re-add them in this commit. Separated to make things easier to
review.
